dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/metrics/BatchMetricsDispatcher.kt

```diff
@@ -123,17 +121,17 @@ internal class BatchMetricsDispatcher(
         )
     }
 
-    private fun resolveFileAge(file: File): Long {
+    private fun File.nameAsTimestampSafe(): Long? {
         return try {
-            dateTimeProvider.getDeviceTimestamp() - file.name.toLong()
+            this.name.toLong()
```

Comment:
we could use `.toLongOrNull()` to avoid throwing an exception

Reply:
I wanted to have the exception just for us though...to see in case we have a weird filename in there. What do you think ?

Reply:
You can still log the filename when the result is null, but throwing an exception costs resources, it's more efficient to avoid it when we can